skbuild/cmaker.py

```diff
@@ -529,7 +529,7 @@ def _guess_python_library(python_version: str) -> str | None:
             candidate_suffixes = [""]
             candidate_implementations = ["python"]
             if hasattr(sys, "pypy_version_info"):
```

Comment:
```suggestion
            if sys.implementation.name == "pypy":
```

In one test, there's a check for `hasattr(sys, "implementation")` too, but I think that's a Python < 3.3 compat leftover.